kqueen/storages/etcd.py

```diff
@@ -592,24 +593,24 @@ def save(self, validate=True, assign_id=True):
             bool: `True` if model was saved without errors, `False` otherwise.
 
         """
+        with etcd.Lock(current_app.db.client, 'customer1'):
+            if assign_id:
+                self.verify_id()
 
-        if assign_id:
-            self.verify_id()
+            validation_status, validation_msg = self.validate()
+            if validate and not validation_status:
+                raise ValueError('Validation for model failed with: {}'.format(validation_msg))
 
-        validation_status, validation_msg = self.validate()
-        if validate and not validation_status:
-            raise ValueError('Validation for model failed with: {}'.format(validation_msg))
+            key = self.get_db_key()
+            logger.debug('Writing {} to {}'.format(self, key))
 
-        key = self.get_db_key()
-        logger.debug('Writing {} to {}'.format(self, key))
+            try:
+                current_app.db.client.write(key, self.serialize())
 
-        try:
-            current_app.db.client.write(key, self.serialize())
-
-            self._key = key
-            return True
-        except Exception:
-            raise
+                self._key = key
+                return True
+            except Exception:
+                raise
```

Comment:
what to raise? maybe we should add generic kqueen exception for it?

Reply:
I just change the indent in scope of this commit, the catched exception is raised again here, but since we do nothing more, I don't know why it is here